tests/envs/test_cooking_env.py

```diff
@@ -28,54 +29,188 @@ def test_cooking_env():
 
     # Uncomment to create video.
     # from gymnasium.wrappers import RecordVideo
+
     # env = RecordVideo(env, "videos/test-cooking-env")
 
     env.action_space.seed(seed)
     obs, _ = env.reset()
     assert isinstance(obs, CookingState)
+    assert obs.meal_temperature is None
 
     # Put the pot on the stove.
-    act = MovePotCookingAction(new_pot_position=(0.5, 0.5))
+    act = MovePotCookingAction(pot_id=0, new_pot_position=(3.0, 6.0))
     obs, reward, terminated, truncated, _ = env.step(act)
     assert np.isclose(reward, 0.0)
     assert not (terminated or truncated)
     assert isinstance(obs, CookingState)
-    assert obs.pot_position == (0.5, 0.5)
+    assert obs.pots[0].position == (3.0, 6.0)
+
+    # Put another pot on the stove
+    act = MovePotCookingAction(pot_id=1, new_pot_position=(3.0, 4.0))
+    obs, reward, terminated, truncated, _ = env.step(act)
+    assert np.isclose(reward, 0.0)
+    assert not (terminated or truncated)
+    assert isinstance(obs, CookingState)
+    assert obs.pots[1].position == (3.0, 4.0)
+
+    # Put another pot on the stove at an occupied location
+    act = MovePotCookingAction(pot_id=2, new_pot_position=(3.0, 5.6))
+    try:
+        obs, reward, terminated, truncated, _ = env.step(act)
+        assert False
+    except ValueError as e:
+        # Check that the error message is correct
+        assert str(e) == "Pot position is already occupied."
+
+    # Put another pot on the stove at an invalid location
+    act = MovePotCookingAction(pot_id=2, new_pot_position=(0.0, 12.0))
+    try:
+        obs, reward, terminated, truncated, _ = env.step(act)
+        assert False
+    except ValueError as e:
+        assert str(e) == "Pot position is outside the stove top."
 
     # Put the ingredient in the pot.
-    act = AddIngredientCookingAction("salt", 0.3)
+    act = AddIngredientCookingAction(pot_id=0, ingredient_id=0, ingredient_quantity=0.3)
+    obs, reward, terminated, truncated, _ = env.step(act)
+    assert np.isclose(reward, 0.0)
+    assert not (terminated or truncated)
+    assert isinstance(obs, CookingState)
+    assert obs.pots[0].ingredient_in_pot_id == 0
+    assert (
+        obs.pots[0].ingredient_in_pot_temperature
+        == scene_spec.ingredients[0].initial_temperature
+    )
+    assert np.isclose(obs.pots[0].ingredient_quantity_in_pot, 0.3)
+
+    # Add to pot not on stove
+    act = AddIngredientCookingAction(pot_id=2, ingredient_id=1, ingredient_quantity=0.1)
+    try:
```

Comment:
see here for a slightly better way to do this kind of testing

https://stackoverflow.com/questions/23337471/how-do-i-properly-assert-that-an-exception-gets-raised-in-pytest